src/TokenFuzzChecks.sol

```diff
@@ -0,0 +1,461 @@
+
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+            privateKey,
+            keccak256(
+                abi.encodePacked(
+                    "\x19\x01",
+                    TokenLike(_token).DOMAIN_SEPARATOR(),
+                    keccak256(abi.encode(PERMIT_TYPEHASH, owner, to, amount, 0, deadline))
+                )
+            )
+        );
+
+        vm.expectEmit(true, true, true, true);
+        emit Approval(owner, to, amount);
+        TokenLike(_token).permit(owner, to, amount, deadline, v, r, s);
+
+        assertEq(TokenLike(_token).allowance(owner, to), amount);
+        assertEq(TokenLike(_token).nonces(owner), 1);
+    }
+
+    function fuzzCheckPermitBadNonce(
+        address _token, 
+        string memory _contractName,
+        uint128 privateKey,
+        address to,
+        uint256 amount,
+        uint256 deadline,
+        uint256 nonce
+    ) public {
+        if (deadline < block.timestamp) deadline = block.timestamp;
+        if (privateKey == 0) privateKey = 1;
+        address owner = vm.addr(privateKey);
+        if (nonce == TokenLike(_token).nonces(owner)) nonce++;
+
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+            privateKey,
+            keccak256(
+                abi.encodePacked(
+                    "\x19\x01",
+                    TokenLike(_token).DOMAIN_SEPARATOR(),
+                    keccak256(abi.encode(PERMIT_TYPEHASH, owner, to, amount, nonce, deadline))
+                )
+            )
+        );
+
+        vm.expectRevert(abi.encodePacked(_contractName, "/invalid-permit"));
+        TokenLike(_token).permit(owner, to, amount, deadline, v, r, s);
+    }
+
+    function fuzzCheckPermitBadDeadline(
+        address _token,
+        string memory _contractName,
+        uint128 privateKey,
+        address to,
+        uint256 amount,
+        uint256 deadline
+    ) public {
+        if (deadline == type(uint256).max) deadline -= 1;
+        if (deadline < block.timestamp) deadline = block.timestamp;
+        if (privateKey == 0) privateKey = 1;
+
+        address owner = vm.addr(privateKey);
+        uint256 nonce = TokenLike(_token).nonces(owner);
+        
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+            privateKey,
+            keccak256(
+                abi.encodePacked(
+                    "\x19\x01",
+                    TokenLike(_token).DOMAIN_SEPARATOR(),
+                    keccak256(abi.encode(PERMIT_TYPEHASH, owner, to, amount, nonce, deadline))
+                )
+            )
+        );
+
+        vm.expectRevert(abi.encodePacked(_contractName, "/invalid-permit"));
+        TokenLike(_token).permit(owner, to, amount, deadline + 1, v, r, s);
+    }
+
+    function fuzzCheckPermitPastDeadline(
+        address _token,
+        string memory _contractName,
+        uint128 privateKey,
+        address to,
+        uint256 amount,
+        uint256 deadline
+    ) public {
+        if (deadline == type(uint256).max) deadline -= 1;
```

Comment:
```suggestion
        vm.assume(deadline < type(uint256).max);
```

Reply:
Changed to using bound() instead as I think that is the preferred method when vm.assume would be excluding a large number of inputs